### PR TITLE
Re-Add the now optional method to automatically analyze crashes

### PR DESCRIPTION
**Pull request by dothebart**
* the user or CI may now specify `--coreCheck true` if the testsuite
  should perform automatic core analysis for him.

This eases the process of evaluating crashes on different targets like windows and mac.

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/

**Comment by jsteemann** [comment]
@dothebart : please resolve the merge conflict

File: js/client/modules/@arangodb/testing.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ let optionsDocumentation = [
   '   - `prefix`:    prefix for the tests in the xml reports',
   '',
   '   - `rr`: if set to true arangod instances are run with rr',
+  '   - `coreCheck`: if set to true we will try to locate coredumps and get a backtrace from it',
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testing.js]: "if set to true, we will attempt to locate a coredump to produce a backtrace in the event of a crash"

File: js/client/modules/@arangodb/testsuites/upgrade.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ function upgrade (options) {
 
   const argv = toArgv(args);
 
-  result.upgrade.first = pu.executeAndWait(pu.ARANGOD_BIN, argv, options, 'upgrade', tmpDataDir);
+  result.upgrade.first = pu.executeAndWait(pu.ARANGOD_BIN, argv, options, 'upgrade', tmpDataDir, options.enableCoreCheck);
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testsuites/upgrade.js]: Isn't this `options.coreCheck` elsewhere?

File: js/client/modules/@arangodb/testsuites/upgrade.js [hunk 3]
```diff
@@ -95,7 +95,7 @@ function upgrade (options) {
 
   ++result.upgrade.total;
 
-  result.upgrade.second = pu.executeAndWait(pu.ARANGOD_BIN, argv, options, 'upgrade', tmpDataDir);
+  result.upgrade.second = pu.executeAndWait(pu.ARANGOD_BIN, argv, options, 'upgrade', tmpDataDir, options.enableCoreCheck);
```
Comment (danielhlarkin) [inline js/client/modules/@arangodb/testsuites/upgrade.js]: Isn't this `options.coreCheck` elsewhere?

**Pull request merged**